internal/server/server.go

```diff
@@ -70,6 +71,11 @@ func New(logger Logger, client Client, templates map[string]*template.Template,
 			allowRoles(client, "System Admin", "Manager")(
 				editTeam(client, templates["team-edit.gotmpl"]))))
 
+	mux.Handle("/teams/delete/",
+		wrap(
+			systemAdminOnly(
```

Comment:
We could have another role with the correct permission, who clicks the button, but is unable to confirm that because they aren't system admin. Is that weird?